internal/moovdata/moov.go

```diff
@@ -27,16 +43,47 @@ func GetTime(path string) (time.Time, error) {
 }
 
 func getTimeFromBoxes(boxes []*mp4.BoxInfoWithPayload) (time.Time, error) {
+	var creationDateIlstIdx *int
+	itemIdx := 0
+
+	var ilstTs *time.Time
+	var mvhdTs *time.Time
 	for _, box := range boxes {
 		switch t := box.Payload.(type) {
-		case *mp4.Meta:
-			// TODO: Parse tags once go-mp4 supports parsing tags from the Meta box
-			// https://github.com/abema/go-mp4/issues/13
+		case *mp4.Keys:
+			for i, k := range t.Entries {
+				if string(k.KeyNamespace) == "mdta" && string(k.KeyValue) == "com.apple.quicktime.creationdate" {
+					// ilst items are not zero indexed so add 1
+					idx := i + 1
+					creationDateIlstIdx = &idx
+				}
+			}
+		case *mp4.Item:
+			itemIdx++
+			if creationDateIlstIdx != nil && itemIdx == *creationDateIlstIdx {
+				ts, err := time.Parse(ilstItemDateLayout, string(t.Data.Data[:]))
+				if err != nil {
+					// unexpected error but swallow in case we find a mvhd timestamp
+					continue
+				}
+				ilstTs = &ts
+			}
 		case *mp4.Mvhd:
-			return timeSince1904(t.CreationTimeV0), nil
+			ts := timeSince1904(t.CreationTimeV0)
+			mvhdTs = &ts
+		default:
 		}
 	}
-	return time.Time{}, errors.New("could not find mvhd box from known mp4 boxes")
+
+	// prefer ilst timestamp
+	if ilstTs != nil {
+		return (*ilstTs).UTC(), nil
+	}
+	if mvhdTs != nil {
+		return (*mvhdTs).UTC(), nil
+	}
```

Comment:
From inspecting a few exported iOS Photos.app media files, I have found that the `ilst` and `mvhd CreationTimestamp` are equal.. I'm not sure when these may differ after probing a few examples..